pkg/drawio/layout.go

```diff
@@ -470,6 +469,26 @@ func (ly *layoutS) setSGLocations() {
 	}
 }
 
+// sortIconsBySGs() sort all the icons by their SGs
+// return a list of lists of icons - all the icons in one list have the same sg
+func sortIconsBySGs(sgs []SquareTreeNodeInterface) [][]TreeNodeInterface {
+	// get all relevant icons:
+	icons := []IconTreeNodeInterface{}
+	for _, sg := range sgs {
+		icons = append(icons, sg.IconTreeNodes()...)
+	}
+	// remove duplicates
+	icons = common.FromList(icons).AsList()
+	// get the icons list for every group of sgs:
+	sgsToIcons := map[setAsKey][]TreeNodeInterface{}
+	for _, icon := range icons {
+		sgsAsKey := icon.SGs().AsKey()
+		sgsToIcons[sgsAsKey] = append(sgsToIcons[sgsAsKey], icon)
+	}
+	// covert to list:
+	return common.MapValues(sgsToIcons)
```

Comment:
Why do we need this extra conversion? Why not return a map?

Reply:
the keys in the map has now meaning.
the desired output of the method should be a list of list.
I think its better coding, thought we pay small run time